lib/bugsnag.js

```diff
@@ -91,7 +91,9 @@ Bugsnag.notify = function(error, options, cb) {
     }
     if (!Bugsnag.shouldNotify()) {
         if (cb) {
-            cb();
+            cb(new Error(!Configuration.apiKey ?
```

Comment:
This should be split up into different errors, just for clarity's sake. I know `shouldNotify` handled the missing API key case, but having them separate is a bit easier to reason about.

```js
if (!Configuration.apiKey) { ... }
else if (!Bugsnag.shouldNotify()) { ... }
```